### PR TITLE
Begin working toward Python3 compatibility.

### PR DESCRIPTION
**Pull request by robertnishihara**

### CODE REVIEW

File: examples/lbfgs/driver.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
```
Comment (mehrdadn) [inline examples/lbfgs/driver.py]: Similarly here, this doesn't seem necessary for a plain `print()`... you'd need it if you're passing an output stream or something though.

**Comment by robertnishihara** [comment]
They aren't "necessary", but they force you to write print statements,
imports, and division in a way that behaves identically in Python 2 and
Python 3. So it's a good practice to include them at the top of every
Python file (tensorflow does this).

This way writing

print "hi"

is an error.

**Comment by mehrdadn** [comment]
Gotcha -- I didn't realize you had actually added them to the top of every file. Cool, nice!

**Comment by robertnishihara** [comment]
This is just the easy stuff. The changes to the C API will be a bit more work.

**Pull request merged**